msal/build.gradle

```diff
@@ -79,6 +89,59 @@ android {
             def fileName = "${archivesBaseName}-${version}.aar"
             outputFileName = fileName
         }
+
+        def buildVariant = variant.name
+        def testTaskName = "test${buildVariant.capitalize()}UnitTest"
+
+        // Tasks for generating Coverage reports(UnitTest & AndroidTests) for every build variant
+        task "${buildVariant}MsalCoverageReport" (type:JacocoReport, dependsOn: ["$testTaskName", "create${buildVariant.capitalize()}CoverageReport"]) {
+            group = "Code Coverage"
+            description = "Generate Msal Coverage Reports(UnitTest & AndroidTests) on the ${buildVariant.capitalize()}"
+
+            def coverageSourceDirs = "${projectDir}/src/main/java"
+            def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test.*']
+            def javaClasses = fileTree(
+                    dir: "${buildDir}/intermediates/javac/${buildVariant}/classes",
+                    excludes: fileFilter
+            )
+            sourceDirectories.from = files([coverageSourceDirs])
+            classDirectories.from = files([javaClasses])
+            executionData.from = fileTree(dir: "$buildDir", includes: [
+                    "jacoco/${testTaskName}.exec"
+            ])
+            reports {
+                xml.enabled = true
+                html.enabled = true
+            }
```

Comment:
Do we need to do all this? Can't this task just depend on code coverage task for unit and android test? (I think all this task needs to do is to just run those other two tasks)

Reply:
For unit tests, we have to do all of this. The "testLocalDebugUnitTest" task runs all the test, that's it but to generate coverage report we are doing all of this. including source files and raw output of task "testLocalDebugUnitTest" is stored in "testLocalDebugUnitTest.exec" file that's why we use it.

But for Android tests, there is no need to do anything. just run the auto generated task "createLocalDebugCoverageReport" and it will generate the coverage report, that's why parent task depends on this task so that both reports could be generated with execution of one task only.

Reply:
added one generic method to reduce repetition of code.